tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs

```diff
@@ -280,6 +284,89 @@ internal void UpdateAddedProbeInstrumentations(IReadOnlyList<ProbeDefinition> ad
             }
         }
 
+        private static string[] SafeParseSignature(string signature)
+        {
+            if (string.IsNullOrEmpty(signature))
+            {
+                return null;
+            }
+
+            try
+            {
+                return ParseSignature(signature);
+            }
+            catch (Exception e)
+            {
+                Log.Error(e, "Error parsing the signature {Signature}. Fall back to empty signature", signature);
+                return null;
+            }
+        }
+
+        private static string[] ParseSignature(string signature)
+        {
+            if (signature.StartsWith("(") && signature.EndsWith(")"))
+            {
+                signature = signature.Substring(1, signature.Length - 2);
```

Comment:
This code is very allocatey - if we can use the vendored/real `Span<T>` it feels like we could avoid the bulk of the allocations here. How concerned are we about that (i.e. how often will this be called/on the hot path)?

Also, would definitely write those unit tests before doing any such refactoring 😉

Reply:
I'm writing a bunch of unit tests that makes sure the logic works 👍

Reply:
A quick note: eventually the parsed out `signature` have to be array of strings since they are passed to the native side with P/Invoke and I don't want to influence the API of the P/Invoke because of the obvious reasons. So eventually I have to create an array of strings on the heap. I've improved the performance of the code in terms of memory.